### PR TITLE
Email Config

### PR DESCRIPTION
**Pull request by ak15**
Generic Email Configuration 
1. SES (AWS) and Sendgrid have same configurations 

We can use this to generate module and configure email smtp server for different projects

### CODE REVIEW

**Review by akuita — APPROVED** [review]
LGTM, will you release npm package?

**Comment by JoelNguyen1608** [comment]
Hi @ak15 , could you give me an example how to use it? For ex: normal SMTP like Gmail? will we need Encrypted Type? Port?...

**Comment by ak15** [comment]
> LGTM, will you release npm package?

Sorry @akuita  will you release npm package meaning , I have generated protobuf  message  for node/middleman
I will merge to master and update in jitera-automation  only

Will not update in middleman or codegen for js , as of now it is planned for only Ruby but future we can use for other generators too for setting up mail config via Studio itself only 

Also may need for preview app 

Please correct if I misunderstood anything 🙏 



@JoelNguyen1608 

Thanks !

For Port 

Globally its is always port(Outgoing for emails) - 587, do you think we should still add  I am confused in this part

I will add one more data as enum called `authentication` 
We can set `authentication` to   "plain", "login", and "cram_md5"
also I will make **service** as enum 

@JoelNguyen1608  can you help me how to create generate protobuf any command for php

By generate protobuf i mean something like we have for nodejs `yarn protobuf:generate`

**Review by JoelNguyen1608 — COMMENTED** [review]
@ak15 

I agreed that 587 is most used. But still some exception, some website I did before, the client using their own STMP server, then they use 25..., so it would be nice if we could have it optional (587 is default or pre-fill)

========
About the generate proto
I cant install it on my laptop (something is wrong with it, bc i'm using WSL2 Ubuntu on Windows 10, still wait for the MacBook) yet, so actually i nerver made it work before
Hi @zrg-team  could you please help on this first? Thank you

**Comment by JoelNguyen1608** [comment]
@ak15 

I only have this https://protobuf.dev/reference/php/php-generated/,  hope it could help, sorry about that :(

**Comment by ak15** [comment]
@JoelNguyen1608 @zrg-team 

When executing command 


`protoc --proto_path=src --php_out=src/GPBMetadata/ src/payload.proto`

I am getting binary files (many different files in php) , its right ????

All  are new binary files so bit confused

File: middleman/lib/protos/v1/payloads.proto [hunk 1]
```diff
@@ -171,12 +171,36 @@ message Feature {
...
+		LOGIN = 1;
+		CRAM_MD5 = 2;
+	}
+
+	enum EmailServiceProvider {
+		AWS_SES = 0;
+		SENDGRID = 1;
+		GMAIL = 2;
+		MAILTRAP = 3;
+	}
```
Comment (JoelNguyen1608) [inline middleman/lib/protos/v1/payloads.proto:202]: Hi @ak15, I'm sorry, but could we have 1 more option "Generic SMTP"? Just incase that user didnit use any of above service?

**Pull request merged**